Floofbot/Modules/Filter.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using Discord;
+using System.Threading.Tasks;
+using Discord.Commands;
+using System.Text.RegularExpressions;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using System.Linq;
+using Discord.Addons.Interactive;
+using Discord.WebSocket;
+using System.Collections.Generic;
+
+namespace Floofbot.Modules
+{
+    [Group("filter")]
+    [Alias("f")]
+    [RequireContext(ContextType.Guild)]
+    [RequireUserPermission(GuildPermission.BanMembers)]
+    [Summary("Settings that control the automatic word filtering in a server")]
+    public class Filter : InteractiveBase
+    {
+        private static readonly Discord.Color EMBED_COLOR = Color.Magenta;
+        private static readonly int WORDS_PER_PAGE = 50;
+        private FloofDataContext _floofDb;
+        public Filter(FloofDataContext floofDb)
+        {
+            _floofDb = floofDb;
+        }
+        private void CheckServerEntryExists(ulong server)
+        {
+            // checks if server exists in database and adds if not
+            var serverConfig = _floofDb.FilterConfigs.Find(server);
+            if (serverConfig == null)
+            {
+                _floofDb.Add(new FilterConfig
+                {
+                    ServerId = server,
+                    IsOn = false
+                });
+                _floofDb.SaveChanges();
+            }
+        }
+        private bool CheckWordEntryExists(string word, SocketGuild guild)
+        {
+            // checks if a word exists in the filter db
+            bool wordEntry = _floofDb.FilteredWords.AsQueryable().Where(w => w.ServerId == guild.Id).Where(w => w.Word == word).Any();
+            return (wordEntry == true) ? true : false;
+        }
+        [Command("toggle")]
+        [Summary("Toggles the word filter")]
+        public async Task Toggle()
+        {
+            await Context.Channel.SendMessageAsync("", false, new EmbedBuilder { Description = $"💾 Usage: `filter toggle channel/server`", Color = EMBED_COLOR }.Build());
+        }
+
+        [Command("toggle")]
+        [Summary("Toggles the word filter")]
+        public async Task Toggle([Summary("channel/server")]string toggleType)
```

Comment:
would be clearer here if the parameter summary was "either one of 'channel' or 'server'", and the method summary was "Toggles the server/channel-level word filter"